src/node/rpc/rpc_context_impl.h

```diff
@@ -16,6 +16,8 @@ namespace ccf
   protected:
     std::shared_ptr<SessionContext> session;
 
+    void* response_user_data;
```

Comment:
The problem with a `void*` is that we can't delete this when the `RpcContext` goes out of scope (eg - the handler calls `set_response_user_data`, then throws). If we instead store this as a `std::shared_ptr<void>`, we get both lifetime management, and a per-instance deleter func, "for free". Example here: https://coliru.stacked-crooked.com/a/c03618118aad2854

Reply:
Yep, that makes sense. I've updated it to be a `shared_ptr<void>`. Unfortunately we can't seem to have templating on the get/set functions, trying it I got linker errors presumably since the setter and getter can't be `virtual` anymore.